src/libs/ajax/GoogleStorage.ts

```diff
@@ -113,6 +125,16 @@ export type GCSListObjectsResponse = {
 
 const encodeAnalysisName = name => encodeURIComponent(`notebooks/${name}`)
 
+interface GoogleFileRaw {
```

Comment:
The GCSItem type already exists for items returned by the GCS API. Could a type like AnalysisGCSItem extend it with the more specific metadata type?

Reply:
I added it as an Optional param to `GcsItem` rather than making it a sub-type, since its optional even for an `AnalysisGCSItem`

Reply:
I like adding `metadata` to GCSItem, but I think for GCSItem it would need to be a very general type like `{ [key: string]: string }`. A type specific to analysis files could extend GCSItem with a more specific type for metadata.

Reply:
I like the idea of making it `abstract`, in declaring it as a union of types instead of a generic set of (key,values). That way, its still generic in `type GCSItem`. Then any consumers can define their own type for it and a predicate to `match` for their type.